sdk/tables/azure-data-tables/tests/test_table_batch.py

```diff
@@ -146,9 +161,14 @@ def _assert_updated_entity(self, entity):
         self.assertEqual(entity.birthday, datetime(1991, 10, 4, tzinfo=tzutc()))
         self.assertFalse(hasattr(entity, "other"))
         self.assertFalse(hasattr(entity, "clsid"))
-        self.assertIsNotNone(entity.odata['etag'])
-        self.assertIsNotNone(entity.timestamp)
-        self.assertIsInstance(entity.timestamp, datetime)
+        self.assertIsNotNone(entity['_metadata']['etag'])
+
+    def _assert_valid_batch_transaction(self, transaction):
+        self.assertIsInstance(transaction, BatchTransactionResult)
+        length = len(transaction.entities)
+        self.assertEqual(length, len(transaction.entities))
```

Comment:
This check seems a bit redundant given that it's how we got the `length` value.
Maybe we could pass the expected length into the method based on the number of operations we added to the batch?